src/java.base/share/classes/java/security/PEMEncoder.java

```diff
@@ -0,0 +1,358 @@
+                        PEMRecord.ENCRYPTED_PRIVATE_KEY, epki.getEncoded()));
+                } catch (IOException e) {
+                    throw new SecurityException(e);
+                }
+            }
+            case X509Certificate c -> {
+                try {
+                    if (isEncrypted()) {
+                        throw new IllegalArgumentException("Certificates " +
+                            "cannot be encrypted");
+                    }
+                    yield pemEncoded(new PEMRecord(PEMRecord.CERTIFICATE,
+                        c.getEncoded()));
+                } catch (CertificateEncodingException e) {
+                    throw new IllegalArgumentException(e);
+                }
+            }
+            case X509CRL crl -> {
+                try {
+                    if (isEncrypted()) {
+                        throw new IllegalArgumentException("CRLs cannot be " +
+                            "encrypted");
+                    }
+                    yield pemEncoded(new PEMRecord(PEMRecord.X509_CRL,
+                        crl.getEncoded()));
+                } catch (CRLException e) {
+                    throw new IllegalArgumentException(e);
+                }
+            }
+            default -> throw new IllegalArgumentException("PEM does not " +
+                "support " + de.getClass().getCanonicalName());
+        };
+    }
+
+    /**
+     * Encodes a given {@code DEREncodable} into PEM.
+     *
+     * @param de the object that implements DEREncodable.
+     * @return a PEM encoded byte[] of the given DEREncodable.
+     * @throws IllegalArgumentException when the passed object returns a null
+     * binary encoding. An exception is thrown when PEMEncoder is
+     * configured for encryption while encoding a DEREncodable that does
+     * not support encryption.
+     * @throws NullPointerException when object passed is null.
+     * @see #withEncryption(char[])
+     */
+    public byte[] encode(DEREncodable de) {
+        return encodeToString(de).getBytes(StandardCharsets.ISO_8859_1);
+    }
+
+    /**
+     * Returns a new immutable PEMEncoder instance configured to the default
+     * encryption algorithm and a given password.
+     *
+     * <p> Only {@link PrivateKey} will be encrypted with this newly configured
+     * instance.  Other {@link DEREncodable} classes that do not support
+     * encrypted PEM will cause encode() to throw an IllegalArgumentException.
+     *
+     * @implNote Default algorithm defined by Security Property {@code
+     * jdk.epkcs8.defaultAlgorithm}.  To configure all the encryption options
+     * see {@link EncryptedPrivateKeyInfo#encryptKey(PrivateKey, char[], String,
+     * AlgorithmParameterSpec, Provider)} and use the returned object with
+     * {@link #encode(DEREncodable)}.
+     *
+     * @param password sets the encryption password.  The array is cloned and
+     *                stored in the new instance.
```

Comment:
Can password be empty? I vaguely remember some algorithms might not like it, or, is it just that `SecretKeySpec` does not like an empty key?